var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -2,231 +2,278 @@
+        "x86_64": "",
+        "thunderx2": "THUNDERX2",
+        "k10": None,
+        "zen": "RYZEN",
+        "bulldozer": "",
+        "piledriver": "",
+        "zen2": "RYZEN",
+        "steamroller": "",
+        "excavator": "",
+        "a64fx": "",
+        "power7": "POWER7",
+        "power8": "POWER8",
+        "power9": "POWER9",
+        "power8le": "POWER8",
+        "power9le": "POWER9",
+        "i686": None,
+        "pentium2": None,
+        "pentium3": None,
+        "pentium4": None,
+        "prescott": None,
+        "nocona": None,
+        "nehalem": None,
+        "sandybridge": "SNB",
+        "haswell": "HSW",
+        "mic_knl": "KNL",
+        "cannonlake": "SKX",
+        "cascadelake": "SKX",
+        "westmere": "WSM",
+        "core2": None,
+        "ivybridge": "SNB",
+        "broadwell": "BDW",
+        "skylake": "SKX",
+        "icelake": "SKX",
+        "skylake_avx512": "SKX",
+    }
+
+    spack_cuda_arch_map = {
+        "30": 'kepler30',
+        "32": 'kepler32',
+        "35": 'kepler35',
+        "37": 'kepler37',
+        "50": 'maxwell50',
+        "52": 'maxwell52',
+        "53": 'maxwell53',
+        "60": 'pascal60',
+        "61": 'pascal61',
+        "70": 'volta70',
+        "72": 'volta72',
+        "75": 'turing75',
+    }
+    cuda_arches = spack_cuda_arch_map.values()
+
+    arch_values = list(arch_variants.keys())
+    allowed_arch_values = arch_values[:]
+    for arch in arch_values:
+        for cuda_arch in cuda_arches:
+            if cuda_arch in arch:
+                conflicts("+%s" % arch, when="~cuda",
+                          msg="Must specify +cuda for CUDA backend to use "
+                              "GPU architecture %s" % arch)
+        dflt, desc = arch_variants[arch]
+        variant(arch, default=dflt, description=desc)
+
+    devices_values = list(devices_variants.keys())
+    for dev in devices_variants:
+        dflt, desc = devices_variants[dev]
+        variant(dev, default=dflt, description=desc)
+
+    options_values = list(options_variants.keys())
+    for opt in options_values:
+        if "cuda" in opt:
+            conflicts('+%s' % opt, when="~cuda",
+                      msg="Must enable CUDA to use %s" % opt)
+        dflt, desc = options_variants[opt]
+        variant(opt, default=dflt, description=desc)
+
+    tpls_values = list(tpls_variants.keys())
+    for tpl in tpls_values:
+        dflt, desc = tpls_variants[tpl]
+        variant(tpl, default=dflt, description=desc)
+        depends_on(tpl, when="+%s" % tpl)
+
+    variant("wrapper", default=False,
+            description="Use nvcc-wrapper for CUDA build")
+    depends_on("kokkos-nvcc-wrapper", when="+wrapper")
+    conflicts("+wrapper", when="~cuda")
+
+    variant("std", default="11", values=["11", "14", "17", "20"], multi=False)
+    # nvcc does not currently work with C++17 or C++20
+    conflicts("+cuda", when="+wrapper std=17")
+    conflicts("+cuda", when="+wrapper std=20")
```

Comment:
Probably don't need `+wrapper` here.

Reply:
Oops. Fixed. Thank you.

Reply:
still there, it seems

Reply:
Not for the last time on our VPN, the git push didn't go through. Nooowwww it's fixed.